common/db.go

```diff
@@ -94,6 +101,13 @@ var schema = `
 	);
 
 	CREATE INDEX IF NOT EXISTS idx_winningtickets_sessionid ON winningTickets(sessionID);
+
+	CREATE TABLE IF NOT EXISTS blockheaders (
+		number int64 PRIMARY KEY,
```

Comment:
If we make `hash` the primary key (see my comment below), then we should add an index for `number` to speed up the `findLatestMiniHeader` and `findAllMiniHeadersSortedByNumber` queries.

Reply:
Will the queries automatically make use of the index?

Is the efficiency gain here for sorting the blocknumber in descending order on those queries?

Reply:
Found an answer

> When you use ORDER BY without an index on the column to be sorted, SQLite needs to build up a temporary data structure that contains the sorted results each time the query is executed. That data structure will use up space in memory (or on disk, depending on the circumstances) and will be torn down after the query is executed. This can be a lot of extra work that might noticeably degrade performance.

[Squeezeing performance from SQLite indexes: Indexes? Indexes!](https://medium.com/@JasonWyatt/squeezing-performance-from-sqlite-indexes-indexes-c4e175f3c346)

Reply:
> Will the queries automatically make use of the index?

Yep.

> Is the efficiency gain here for sorting the blocknumber in descending order on those queries?


Yaeh - adding an index can help avoid the sort because the index can allow the [rows to be delivered in the required order](https://use-the-index-luke.com/sql/sorting-grouping/indexed-order-by).